### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by Pessimistress**
- migrate source from deck.gl
- migrate tests from deck.gl
- add main example app

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "plugins": [
+    "react"
+  ],
+  "extends": [
+    "eslint-config-uber-jsx",
```
Comment (ibgreen) [inline .eslintrc]: Maybe keep the jsx config in `examples`

File: examples/main/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const config = {
+  entry: {
+    app: resolve('./app.js')
+  },
+
+  devtool: 'source-map',
+
+  module: {
+    rules: [{
+      // Compile ES2015 using bable
```
Comment (ibgreen) [inline examples/main/webpack.config.js]: typo: Babel

File: examples/main/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// NOTE: To use this example standalone (e.g. outside of deck.gl repo)
```
Comment (ibgreen) [inline examples/main/webpack.config.js]: cut `deck.gl` (better do a global search).

File: examples/webpack.config.local.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// This file contains webpack configuration settings that allow
+// examples to be built against the deck.gl source code in this repo instead
+// of building against their installed version of deck.gl.
```
Comment (ibgreen) [inline examples/webpack.config.local.js]: Cut deck.gl

File: package.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    "lint": "eslint src examples test",
+    "test": "npm run lint && node test/node.js",
+    "publish-prod": "npm run build && npm run test && npm publish",
+    "publish-beta": "npm run build && npm run test && npm publish --tag beta"
+  },
+  "pre-commit": [
+    "test"
+  ],
+  "dependencies": {
+    "hammerjs": "^2.0.8"
```
Comment (ibgreen) [inline package.json]: Hmm these guys are called `hammerjs` not `hammer.js` I guess we should have been `mjolnirjs`. But that looks unreadable...

File: package.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  "module": "dist-es6/index.js",
+  "files": [
+    "src",
+    "dist"
+  ],
+  "scripts": {
+    "start": "(cd examples/main && (path-exists node_modules || npm i) && npm run start-local)",
+    "build": "npm run clean && npm run build-es5 && npm run build-es6",
+    "build-es5": "mkdir -p dist && babel src --out-dir dist --plugins=transform-es2015-modules-commonjs,transform-runtime --source-maps inline",
+    "build-es6": "mkdir -p dist-es6 && babel src --out-dir dist-es6 --source-maps inline",
```
Comment (ibgreen) [inline package.json]: We should stop inlining source maps - but had some problems when I tried and never got around to completing it.

File: src/event-manager.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+      const {element} = this;
+      const {srcEvent} = event;
+
+      const center = event.center || {
+        x: srcEvent.clientX,
+        y: srcEvent.clientY
+      };
+
+      // Calculate center relative to the root element
+      // TODO/xiaoji - avoid using getBoundingClientRect for perf?
```
Comment (ibgreen) [inline src/event-manager.js]: Remove real names in open source?

File: .eslintrc (hunk 1 above)
Comment (Pessimistress) [reply .eslintrc]: We don't config lint in individual examples

**Pull request merged**